raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -765,18 +840,16 @@ private void handleInboundMessage(RaftMessage message, long currentTimeMs) throw
         } else if (message instanceof RaftResponse.Inbound) {
             RaftResponse.Inbound response = (RaftResponse.Inbound) message;
             logger.debug("Received response from {}: {}", response.sourceId(), response.data);
-            boolean responseHandled = maybeHandleError(response, currentTimeMs);
-            if (!responseHandled)
-                handleResponse(response);
+            handleResponse(response, currentTimeMs);
         } else {
-            throw new IllegalStateException("Unexpected message " + message);
+            throw new IllegalArgumentException("Unexpected message " + message);
```

Comment:
Why change it to illegal-argument? I think illegal-state is more accurate :P

Reply:
The difference between these two exceptions can get confusing in my mind. Here I see this as basically a problem of an invalid input, so I thought `IllegalArgumentException` was a better fit.